doc: nrf5340audio: add docs for new apps (#37)
alexsven/sdk-nrf: applications/nrf5340_audio/doc/firmware_architecture.rst

   :local:
   :depth: 2

> Each nRF5340 Audio application corresponds to one specific LE Audio role: gateway or headset.

Reviewer: "Gateway" and "Headset" is not LE Audio Roles.
Suggest: 
```suggestion
Each nRF5340 Audio application corresponds to one specific LE Audio role: unicast client (gateway), unicast server (headset), broadcast source (gateway) or broadcast sink (headset).
```

> The gateway receives the audio data from external sources (USB or I2S) and forwards it to one or more headsets.

Reviewer: ```suggestion
The gateway receives the audio data from external sources (USB or line input/I2S) and forwards it to one or more headsets.
```

> The headset is a receiver device that plays back the audio it gets from the gateway, for example earbuds, headphones, or hearing aid.

Reviewer: ```suggestion
The headset is a receiver device that plays back the audio it gets from the gateway, and will act as for example earbuds, headphones, a speaker or hearing aids.
```

> Similarly, each nRF5340 Audio application is configured for one specific LE Audio mode: the *connected isochronous stream* (CIS) mode or in the *broadcast isochronous stream* (BIS) mode.

Reviewer: ```suggestion
Each nRF5340 Audio application is configured for one specific LE Audio mode: the *connected isochronous stream*/unicast (CIS) mode or in the *broadcast isochronous stream* (BIS) mode.
```

See :ref:`nrf53_audio_app_overview_modes` for more information.

> All the applications use the same code base, except different :file:`main.c` files.

Reviewer: IMO; we should phrase it more general: 
```suggestion
The applications use the same code base, but uses different :file:`main.c` files and includes different modules and libraries depending on the configuration.
```

You might need to configure and program two applications for testing the interoperability, depending on your use case.
See :ref:`nrf53_audio_app_tutorial_cis` and :ref:`nrf53_audio_app_tutorial_bis` for more information.

.. _nrf53_audio_app_overview_differences:

Differences between apps
************************

The following table summarizes the differences between the available nRF5340 Audio applications.

.. list-table:: Differences between nRF5340 Audio applications
   :header-rows: 1

   * - Application
     - LE Audio mode and role
   * - :ref:`Unicast client<nrf53_audio_unicast_client_app>`
     - CIS gateway
   * - :ref:`Unicast server<nrf53_audio_unicast_server_app>`
     - CIS headset
   * - :ref:`Broadcast sink<nrf53_audio_broadcast_sink_app>`
     - BIS gateway
   * - :ref:`Broadcast source<nrf53_audio_broadcast_source_app>`
     - BIS headset

> If you :ref:`select the I2S serial connection<nrf53_audio_app_configuration_select_i2s>` to enable the `I2S-based firmware for gateway and headsets`_, sending the audio data through the application layers includes a mandatory synchronization step using the synchronization module.

Reviewer: Even though USB is used, we still need to have this sync module in place on the headset side.

Author: Meaning the file is there also when we use USB?

Author: Removed the line during the meeting.

* Application-specific custom modules:

  * FIFO buffers
  * Synchronization module (part of `I2S-based firmware for gateway and headsets`_) - See `Synchronization module overview`_ for more information.

> Since the application architecture is the same for all applications and the code before compilation is shared to a significant degree, the set of modules in use depends on the chosen audio inputs and outputs (USB or analog jack).

Reviewer: Above: Suggest we remove the list of custom modules.

Author: Would keeping the synchronization module acceptable?

.. note::
   In the current versions of the applications, the bootloader is disabled by default.
   Device Firmware Update (DFU) can only be enabled when :ref:`nrf53_audio_app_building_script`.
   See :ref:`nrf53_audio_app_configuration_configure_fota` for details.

Communication between modules
=============================

Communication between modules is primarily done through Zephyr's :ref:`zephyr:zbus` to make sure that there are as few dependencies as possible. Each of the buses used by the applications has their message structures described in :file:`nrf5340_audio_common.h`.

> The applications use the following buses:

Reviewer: Suggest we remove this as well. Which channels are used may depend on the application. This is also quite technical.

Author: Kept line 137.

USB-based firmware for gateway
==============================

The following figures show an overview of the modules currently included in the firmware of applications that use USB.

> In this firmware design, no synchronization module is used after decoding the incoming frames or before encoding the outgoing ones.

Reviewer: @alexsven : Discuss.

Author: Discussed and so far it is correct.
